### PR TITLE
Add latest changes from staticweb core

### PR DESCRIPTION
**Pull request by StefanWallin**
This PR is intended to review the current unpublished changes that reside in the https://github.com/StaticWebCMS/core project.

### CODE REVIEW

File: jStorage.github.js [hunk 1]
```diff
@@ -58,49 +33,97 @@
...
-                    try {
-                        self._config.callback(wrapper, callStatus);
-                    } catch (e) {
-                        // TODO: handle error
-                    }
-                }
+            this.ensureAuth(wrapper, config);
+
+            // https://github.com/settings/tokens/new
+            var token = this._config.token;
```
Comment (StefanWallin) [inline jStorage.github.js]: this line is not self explanatory. Don't understand why it's there, what is it doing?

Comment (StefanWallin) [inline jStorage.github.js]: It also does not seem like the token variable is ever used in this function.

File: jStorage.github.js [hunk 2]
```diff
@@ -310,67 +334,105 @@
...
-            this._hasToken = config && typeof (config.token) === "string";
-            this._hasTokenService = config && typeof (config.tokenService) === "string";
-            if (this._hasTokenService && !this._hasToken) {
-                // TODO: We should probably make sure that the returned scopes is equal to our requested scope
-                var tmpToken = this.getUrlParameterByName('access_token', window.location);
-                if (!!tmpToken) {
-                    config.token = tmpToken;
-                    this._hasToken = true;
-                }
+            var token = window.localStorage.getItem('token');
```
Comment (StefanWallin) [inline jStorage.github.js]: Maybe add some comment here saying: `If we have a token stored in localStorage, we can use that and ignore the rest below`

File: jStorage.dropbox.js [hunk 3]
```diff
@@ -79,6 +79,13 @@
 
             });
         },
+        listStorages: function (callback) {
```
Comment (StefanWallin) [inline jStorage.dropbox.js]: do we need to add this function to other providers as well? jStorage.localStorage for example.

File: jStorage.github.js [hunk 4]
```diff
@@ -58,49 +33,97 @@
...
+                }
+            });
         },
-        get: function (name, callback) {
+        get: function (name, callback, extendedOptions) {
             var self = this;
 
+            var repo = this.getRepo(this._config.repo, extendedOptions);
+
             // Remove begining slash
```
Comment (flowertwig-org) [inline jStorage.github.js]: fixed in comming commit

File: jStorage.github.js (hunk 1 above)
Comment (flowertwig-org) [inline jStorage.github.js]: fixed in comming commit

File: jStorage.github.js [hunk 5]
```diff
@@ -205,7 +227,7 @@
             //if (name && name[name.length - 1] == '/') {
```
Comment (flowertwig-org) [inline jStorage.github.js]: fixed in coming commit :)

File: jStorage.github.js [hunk 6]
```diff
@@ -280,8 +300,10 @@
             var self = this;
             addr = "https://api.github.com/repos/" + this._config.repo + "/contents/" + name;
             githubRequest("GET", addr, self._config.token, false, function () {
+                console.log('listA', arguments);
```
Comment (flowertwig-org) [inline jStorage.github.js]: fixed in comming commit

File: jStorage.github.js [hunk 7]
```diff
@@ -280,8 +300,10 @@
...
                 if (arguments.length >= 2) {
                     var info = arguments[1];
+                    console.log('list', info);
```
Comment (flowertwig-org) [inline jStorage.github.js]: fixed in comming commit

File: jStorage.github.js [hunk 8]
```diff
@@ -296,6 +318,8 @@
                             self._shaCache[info[i].path] = info[i].sha;
                         }
 
+                        console.log('list content:', list);
```
Comment (flowertwig-org) [inline jStorage.github.js]: fixed in comming commit

File: jStorage.github.js [hunk 9]
```diff
@@ -310,67 +334,105 @@
...
-            if (!url) url = window.location.href;
-            name = name.replace(/[\[\]]/g, "\\$&");
-            var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
-                results = regex.exec(url);
-            if (!results) return null;
-            if (!results[2]) return '';
-            return decodeURIComponent(results[2].replace(/\+/g, " "));
+        getTokenFromQuery: function () {
+            // remove questionmark from querystring
+            var search = window.location.search.substr(1);
```
Comment (flowertwig-org) [inline jStorage.github.js]: also adjusted in comming commit

File: jStorage.github.js [hunk 10]
```diff
@@ -310,67 +334,105 @@
...
+            var arr = search.split('&');
```
Comment (flowertwig-org) [inline jStorage.github.js]: adjusted in comming commit

File: jStorage.github.js [hunk 11]
```diff
@@ -310,67 +334,105 @@
...
+                    }
+                }
+            }
+            return false;
         },
         ensureAuth: function (wrapper, config) {
-            // https://github.com/settings/tokens/new 
+            // we require token today, allow for oauth fetching of token here
+
+            // Because of security reasons, if we have valid token in url, force remove...
```
Comment (flowertwig-org) [inline jStorage.github.js]: fixed in comming commit

File: jStorage.github.js (hunk 2 above)
Comment (flowertwig-org) [inline jStorage.github.js]: good call, will be added in next commit :)

File: jStorage.dropbox.js (hunk 3 above)
Comment (flowertwig-org) [inline jStorage.dropbox.js]: ops, yes of course.

File: jStorage.github.js (hunk 1 above)
Comment (flowertwig-org) [inline jStorage.github.js]: all github related in this pr has been fixed in a refactoring.
Can be viewed at #12

**Comment by flowertwig-org** [comment]
Everything in this PR + more is now included in other PR:s.
Agree that we can close this one? :)

**Pull request closed without merging**